Sources/FirebaseFirestore/DocumentSnapshot+Swift.swift

```diff
@@ -9,9 +9,37 @@ import CxxShim
 import Foundation
 
 public typealias DocumentSnapshot = firebase.firestore.DocumentSnapshot
+public typealias ServerTimestampBehavior = firebase.firestore.DocumentSnapshot.ServerTimestampBehavior
+public typealias MapFieldValue = firebase.firestore.MapFieldValue
+
+public typealias GeoPoint = firebase.firestore.GeoPoint
+public typealias FieldValue = firebase.firestore.FieldValue
```

Comment:
All of these types are used in some way with the new codable APIs and the parsing code in general so we needed to bridge them to the correct type names for Swift to work.

Reply:
If these are `public`, we should ensure that these match the firebase API.

Reply:
I can probably make `MapFieldValue` not public since that doesn't have a matching type in the iOS SDK, but GeoPoint and FieldValue both end up having Objective-C classes that bridge them into Swift for iOS so I think we should leave those exposed as `public` and add the APIs we required as we need.